src/vec/include/packed4.hpp

```diff
@@ -0,0 +1,86 @@
+/*
+  Copyright (C) 2021
+      Max Planck Institute for Polymer Research & JGU Mainz
+
+  This file is part of ESPResSo++.
+
+  ESPResSo++ is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  ESPResSo++ is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef VEC_INCLUDE_PACKED4_HPP
+#define VEC_INCLUDE_PACKED4_HPP
+
+#include "Real3D.hpp"
+
+namespace espressopp { namespace vec {
+
+  struct Real3DInt {
+    real x;
+    real y;
+    real z;
+    lint t;
+
+    Real3DInt()
+    {
+      // for padding to work, real must be 64 bit floating point type
+      static_assert(sizeof(real)==sizeof(lint),
```

Comment:
Seems odd to check the size of a floating point type against the size of an integer type. Why not `sizeof(double)`.

This check will de facto eliminate the possibility to build in single precision rendering the choice irrelevant.

Reply:
Will be removing the AOS implementation so this part won't be needed.